### PR TITLE
feat(aws-dynamodb-global): Adding package to support DynamoDB Global Tables

### PR DESCRIPTION
**Pull request by KingOfPoptart**
Adding a new package to support AWS DynamoDB Global Tables.  The new package is built on top the existing aws-dynamodb package.

The construct creates multiple stacks in the specified regions, each with a DynamoDB Table (using the aws-dynamodb package).  A custom resource is then deployed and uses a lambda to tie the tables together into a multi-region, multi-master database.

----

### Pull Request Checklist

* [x] Testing
  - Unit test added (prefer not to modify an existing test, otherwise, it's probably a breaking change)
  - __CLI change?:__ coordinate update of integration tests with team
  - __cdk-init template change?:__ coordinated update of integration tests with team
* [x] Docs
  - __jsdocs__: All public APIs documented
  - __README__: README and/or documentation topic updated
* [x] Title and Description
  - __Change type__: title prefixed with **fix**, **feat** will appear in changelog
  - __Title__: use lower-case and doesn't end with a period
  - __Breaking?__: last paragraph: "BREAKING CHANGE: <describe what changed + link for details>"
  - __Issues__: Indicate issues fixed via: "**Fixes #xxx**" or "**Closes #xxx**"
* [x] Sensitive Modules (requires 2 PR approvers)
  - IAM Policy Document (in @aws-cdk/aws-iam)
  - EC2 Security Groups and ACLs (in @aws-cdk/aws-ec2)
  - Grant APIs (only if not based on official documentation with a reference)

----

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license.

### CODE REVIEW

**Comment by KingOfPoptart** [comment]
So for clarity - I completely screwed up my previous PR (rebasing nightmare from waiting 20+ days between starting and getting ready to merge), so I'm creating this one to replace it.  Old PR can be found here - https://github.com/awslabs/aws-cdk/pull/2082

File: packages/@aws-cdk/aws-dynamodb-global/lib/lambda-global-dynamodb.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import lambda = require("@aws-cdk/aws-lambda");
+import cdk = require("@aws-cdk/cdk");
+import path = require("path");
+import { GlobalTableProps } from "./aws-dynamodb-global";
+
+/**
+ * A stack that will make a Lambda that will launch a lambda to glue
+ * together all the DynamoDB tables into a global table
+ */
+export class LambdaGlobalDynamoDBMaker extends cdk.Stack {
```
Comment (eladb) [inline packages/@aws-cdk/aws-dynamodb-global/lib/lambda-global-dynamodb.ts]: Sorry for nitpicking, I am a bit sensitive to naming. Can you rename this to `GlobalTableCoordinator` (it's an internal class, I know...).

File: packages/@aws-cdk/aws-dynamodb-global/lib/lambda-global-dynamodb.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      runtime: lambda.Runtime.NodeJS810,
+      timeout: 300,
+      uuid: "D38B65A6-6B54-4FB6-9BAD-9CD40A6DAC12",
+    });
+    LambdaGlobalDynamoDBMaker.grantCreateGlobalTableLambda(this.lambdaFunction.role);
+    this.customResource = new cfn.CustomResource(this, "CfnCustomResource", {
+      lambdaProvider: this.lambdaFunction,
+      properties: {
+        regions: props.regions,
+        resourceType: "Custom::MakeGlobalDynamoDB",
```
Comment (eladb) [inline packages/@aws-cdk/aws-dynamodb-global/lib/lambda-global-dynamodb.ts]: Rename to `Custom::DynamoGlobalTableCoordinator`

File: packages/@aws-cdk/aws-dynamodb-global/package.json [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+      "dotnet": {
+        "namespace": "Amazon.CDK.AWS.DynamoDB.Global",
+        "packageId": "Amazon.CDK.AWS.DynamoDB.Global",
+        "signAssembly": true,
+        "assemblyOriginatorKeyFile": "../../key.snk"
+      },
+      "sphinx": {}
+    }
+  },
+  "cdk-build": {
```
Comment (eladb) [inline packages/@aws-cdk/aws-dynamodb-global/package.json]: Remove this from here... It's only for the aws-dynamodb package

**Comment by eladb** [comment]
@KingOfPoptart do you mind taking a few minutes finalizing the PR title and message so we can use them for the merge commit?

**Comment by KingOfPoptart** [comment]
@eladb updated!  Let me know if that's sufficient.

**Pull request merged**